File: src/main/java/ru/citeck/ecos/uiserv/domain/menu/dto/MenuItemActionDef.java
package ru.citeck.ecos.uiserv.domain.menu.dto;

import lombok.Data;
import ru.citeck.ecos.commons.data.ObjectData;
import ru.citeck.ecos.commons.json.serialization.annotation.IncludeNonDefault;

@Data
@IncludeNonDefault
public class MenuItemActionDef {
    private String type;
    private ObjectData config;
}

File: src/main/java/ru/citeck/ecos/uiserv/domain/menu/dto/MenuItemActionDef.kt
package ru.citeck.ecos.uiserv.domain.menu.dto

import ru.citeck.ecos.commons.data.ObjectData
import ru.citeck.ecos.commons.json.serialization.annotation.IncludeNonDefault

@IncludeNonDefault
data class MenuItemActionDef(
    val type: String,
    val config: ObjectData
) {
    companion object {
        val EMPTY = MenuItemActionDef("", ObjectData.create())
    }
}
